Sets default X-Frame-Options for login and registration forms to environment setting

nizarmah opened the pull request:
Update the `X-Frame-Options` default value for login and registration forms to the existing `EDXAPP_X_FRAME_OPTIONS` environment settings to match the default setting across the whole platform. 

**JIRA tickets**: SE-3437, [OSPR-5053](https://openedx.atlassian.net/browse/OSPR-5053)

**Sandbox URL**:
- **LMS**: https://pr25338.sandbox.opencraft.hosting/
- **Studio**: https://studio.pr25338.sandbox.opencraft.hosting/

**Testing instructions**:

1. Send a request to the Login page
2. Verify that the response header contains `X-Frame-Options: ALLOW`

**Reviewers**
- [ ] @toxinu 
- [ ] @gabor-boros 
- [ ] edX reviewer[s] TBD

**Settings**
```yaml
EDXAPP_X_FRAME_OPTIONS: "ALLOWALL"
```

edx-webhook commented:
Thanks for the pull request, @nizarmah! I've created [OSPR-5053](https://openedx.atlassian.net/browse/OSPR-5053) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

gabor-boros reviewed (approved):
👍  🎉 

- [x] I tested this: Send a request to the Login page with a specific Referer: customhostname.com header
- [x] I tested this: Verify that the response header contains X-Frame-Options: ALLOW
- [x] I read through the code
~- [ ] I checked for accessibility issues~
~- [ ] Includes documentation~
~- [ ] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.~

natabene commented:
@nizarmah Thank you for your contribution. Please let me know once it is ready for our review.

natabene commented:
@nizarmah When do you think this PR will be ready?

nizarmah commented:
@natabene sorry about this PR. It might be a while since it is ready.

There are a couple of changes I'm trying to upstream at this moment, which include different fixes, such as the ones open at the moment to the Video Abstraction Layer repository (edx-val). 

This is PR is one of the changes that I'd like to upstream, but there's a high chance I'll be changing the approach that I used on this PR, which is why I converted it to a Draft PR.

If it would be an issue, I can close it and reopen a new one. Let me know please :slightly_smiling_face:

natabene commented:
@nizarmah Thanks for the update, it is fine to keep it open, just wanted to make sure this was not abandoned.

nizarmah commented on common/djangoapps/third_party_auth/decorators.py:
```diff
@@ -29,7 +29,12 @@ def wrapped_view(request, *args, **kwargs):
-                if LTIProviderConfig.objects.current_set().filter(lti_hostname=hostname, enabled=True).exists():
+
+                if (
+                    (settings.X_FRAME_ALLOW_WHITELISTED_REFERERS
```
Resolving this comment since I changed the approach.

nizarmah commented:
@natabene this is ready for review :+1:

natabene commented:
@nizarmah Thank you for letting me know.

bradenmacdonald reviewed (approved):
Has potential security / clickjacking implications, but I have checked: 
* This change just makes the `X_FRAME_OPTIONS` setting work more consistently as described, and does not affect the security of the default nor production configurations.
* This really only affects IE 11 / old browsers, because a "Content Security Policy" with "frame-ancestors" is used for newer browsers and allows more fine-grained control.
* Tested this on local devstack to confirm `DENY` is still the default header sent.

So: :+1:

- [x] I tested this: as described above and on the sandbox provided
- [x] I read through the code
- [x] I checked for accessibility issues: n/a
- [x] Includes documentation: n/a, setting is already documented

bradenmacdonald commented:
@nizarmah Can you please rebase + squash this down to one commit with a clear commit message, in preparation for merge?

nizarmah commented:
@bradenmacdonald thanks for your review! :smile: 
I've rebased and squashed down to a single commit with a clear commit message :+1: 

I'll keep an eye on the builds, in case they fail (even though they should not)

natabene commented:
@bradenmacdonald Ready to merge?

bradenmacdonald commented:
@natabene Yes, this is ready to merge; I have asked @ormsbee if he can do that.

edx-webhook commented:
@nizarmah 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-platform&entry.867055334=https://github.com/edx/edx-platform/pull/25338&entry.1484655318=https://github.com/nizarmah&entry.752974735=2020-10-13+11:05&entry.1917517419=2020-12-18+17:39&entry.2133058324=Yes) so we can improve your experience in the future.
